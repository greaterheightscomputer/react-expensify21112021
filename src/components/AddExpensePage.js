// import React from "react";
// import { connect } from "react-redux";
// import { addExpense } from "../actions/expenses";
// import ExpenseForm from "./ExpenseForm";

// export class AddExpensePage extends React.Component {
//   onSubmit = (expense) => {
//     this.props.addExpense(expense);
//     this.props.history.push("/");
//   };

//   render() {
//     return (
//       <div>
//         <h1>Add Expense</h1>
//         <ExpenseForm onSubmitForm={this.onSubmit} />
//       </div>
//     );
//   }
// }

// const mapDispatchToProps = (dispatch) => {
//   return {
//     addExpense: (expense) => dispatch(addExpense(expense)),
//   };
// };

// export default connect(undefined, mapDispatchToProps)(AddExpensePage);

//Changing of addExpense() Action Generator Function to startAddExpense() Asynchronous Action
//Generator Function
import React from "react";
import { connect } from "react-redux";
import { startAddExpense } from "../actions/expenses";
import ExpenseForm from "./ExpenseForm";

export class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    this.props.history.push("/");
  };

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm onSubmitForm={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    startAddExpense: (expense) => dispatch(startAddExpense(expense)),
  };
};

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
