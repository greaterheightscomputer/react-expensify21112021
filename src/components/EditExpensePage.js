import React from 'react';
import { connect } from 'react-redux';
import { startEditExpense, startRemoveExpenese } from '../actions/expenses';
import ExpenseForm from './ExpenseForm';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.startRemoveExpenese({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmitForm={this.onSubmit}
        />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

//fetch data from redux store
const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(
      (expense) => expense.id === props.match.params.id
    ),
  };
};

//setup mapDispatchToProps()
const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpenese: (data) => dispatch(startRemoveExpenese(data)),
});
export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
