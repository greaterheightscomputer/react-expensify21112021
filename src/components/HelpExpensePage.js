import React from "react";

const HelpExpensePage = () => (
  <div>This is from my help expenses component.</div>
);

export default HelpExpensePage;
