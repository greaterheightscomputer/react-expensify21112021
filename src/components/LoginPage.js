import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

// export const LoginPage = (props) =>  (
//     <div>
//       <button onClick={props.startLogin}>Login</button>
//     </div>
//   );
//

//destructure props and connect LoginPage to redux store
export const LoginPage = ({ startLogin }) => (
  <div>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
