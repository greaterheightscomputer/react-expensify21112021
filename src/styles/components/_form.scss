.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size; //>* apply these style to the direct children of form class selector
  }
}
