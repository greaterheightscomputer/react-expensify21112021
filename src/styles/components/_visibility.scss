.show-for-mobile {
  //width <=720px or 45rem display Expenses
  @media (min-width: $desktop-breakpoint) {
    display: none;
  }
}

.show-for-desktop {
  //width >=720px or 45rem display Expense and Amount
  //0.01rem means both max-width and min-width never meet
  @media (max-width: $desktop-breakpoint - 0.01rem) {
    display: none;
  }
}
