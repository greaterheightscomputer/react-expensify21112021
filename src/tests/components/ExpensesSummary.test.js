import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should correctly render ExpensesSummary with 1 expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={240} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should correctly render ExpensesSummary with multiple expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={13} expensesTotal={8900} />
  );
  expect(wrapper).toMatchSnapshot();
});
