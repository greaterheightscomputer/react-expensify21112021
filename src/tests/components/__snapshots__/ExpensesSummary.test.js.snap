// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Veiwing 
    1
     
    expense
     totalling 
    ₦2.40
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Veiwing 
    13
     
    expenses
     totalling 
    ₦89.00
  </h1>
</div>
`;
